runtimes/node-18.0/server.js

```diff
@@ -1,85 +1,138 @@
+                }
+            }
+            logs.push(args.join(" "));
+        },
+        error: function () {
+            const args = [];
+            for (const arg of Array.from(arguments)) {
+                if (arg instanceof Object || Array.isArray(arg)) {
+                    args.push(JSON.stringify(arg));
+                } else {
+                    args.push(arg);
+                }
+            }
+            errors.push(args.join(" "));
+        },
     };
 
+    console.log = console.info = console.debug = console.warn = console.error = function() {
+        logs.push('Unsupported log noticed. Use context.log() or context.error() for logging.');
+    }
+
+    let output = null;
     try {
-        let userFunction = require(USER_CODE_PATH + '/' + process.env.INTERNAL_RUNTIME_ENTRYPOINT);
+        let userFunction = require(USER_CODE_PATH + '/' + process.env.OPEN_RUNTIMES_ENTRYPOINT);
 
         if (!(userFunction || userFunction.constructor || userFunction.call || userFunction.apply)) {
-            throw new Error("User function is not valid.")
+            throw new Error("User function is not valid.");
         }
 
-        if(userFunction.default) {
+        if (userFunction.default) {
             if (!(userFunction.default.constructor || userFunction.default.call || userFunction.default.apply)) {
-                throw new Error("User function is not valid.")
+                throw new Error("User function is not valid.");
             }
-            
-            await userFunction.default(request, response);
+
+            output = await userFunction.default(context);
         } else {
-            await userFunction(request, response);
+            output = await userFunction(context);
         }
     } catch (e) {
-        send(res, 500, {stdout: logs.join('\n'), stderr: errors.join('\n') + "\n" + e.code === 'MODULE_NOT_FOUND' ? "Code file not found." : e.stack || e});
+        context.error(e.code === 'MODULE_NOT_FOUND' ? "Code file not found." : e.stack || e);
+        output = context.res.send('', 500);
+    }
+
+    if(output === null || output === undefined) {
+        context.error('Return statement missing. return context.res.empty() if no response is expected.');
+        output = context.res.send('', 500);
+    }
+
+    output.body = output.body ?? '';
+    output.statusCode = output.statusCode ?? 204;
+    output.headers = output.headers ?? {};
+
+    for (const header in output.headers) {
+        if(header.startsWith('x-open-runtimes-')) {
+            continue;
+        }
+        
+        res.setHeader(header, output.headers[header]);
     }
 
-    console.log = console.stdlog;
-    console.error = console.stderror;
-    console.debug = console.stddebug;
-    console.warn = console.stdwarn;
-    console.info = console.stdinfo;
+    res.setHeader('x-open-runtimes-logs', encodeURIComponent(logs.join('\n')));
+    res.setHeader('x-open-runtimes-errors', encodeURIComponent(errors.join('\n')));
```

Comment:
We wanted to support logs streaming at one point. Im not sure headers would be the right way to send logs in that case. Its something to think about

Reply:
Only possibility in HTTP that allows streaming (as far as I am aware) is body. We surely dont want to touch body and define open-runtimes-specific structure.

I believe streaming support can be added in future with a reserved endpoint.

Reply:
Okay. Problem for later 👍